### DISCUSSION TITLE
一些功能期望。

### DISCUSSION THREAD

**Discussion by mieyeaa** [anyproto/contributors#20] [Localization]
1--什么时候可以任意语言化，比如：中文，或其它语言化。
2--安装问题，安装路径可选择，别默认安装在C盘。

**Comment by jidenanian** [comment]
同样关注本地化的问题，希望能加快该进程

**Comment by fuksman** [comment]
Hi @mieyeaa and @jidenanian, thank you for your readiness to participate in the project! 
We are polishing localization files and hopefully will launch the l18n project next week.

**Reply by Xonline-Tech** [reply]
Hi, I have initially completed the localization of the `text.json` and `readme.md` documents in the desktop program [Xonline-Tech](https://github.com/Xonline-Tech)/[anytype-ts-cn](https://github.com/Xonline-Tech/anytype-ts-cn/tree/translate/simplified_chinese), but I found that the language cannot be switched on the front page, and there are a lot of words that have not been added to the i18n adaptation. I hope my contribution will have a chance to be applied, and a more complete i18n adaptation work will be started as soon as possible.Wish better and better

**Reply by ra3orblade** [reply]
@Xonline-Tech hey man, check this branch pls, https://github.com/anyproto/anytype-ts/pull/88, I've wanted to help you so i've started finishing moving all strings to the localisation file, I've got only ~115 files left lol, so I will finish it this week, I can then import your current state of translations by hands without merging it because I have written some tools for it, so then you can continue working on it. You can contact me over telegram if you have any questions.

**Reply by hit-lacus** [reply]
@Xonline-Tech  Hey, it is time to contribute Chinese translation to new repo:  https://github.com/anyproto/l10n-anytype-ts/blob/main/locales/zh-CN.json .

**Reply by fuksman** [reply]
Hey @hit-lacus! While it's safe to start providing translations on Crowdin, I want to point out that we haven't finished extracting strings yet.

**Reply by ra3orblade** [reply]
https://github.com/anyproto/anytype-ts/pull/109/files language switch PR, sorry I didn't add Chinese to the options, was just testing stuff, so you can do it on your own in `src/json/constant.json` -> `"interfaceLangs"`.

**Reply by hit-lacus** [reply]
OK, thanks for message, I didn't realized that, maybe you will announce somewhere when the job 'extracting strings' is done.

**Reply by Xonline-Tech** [reply]
Ok, I will update i18n for Simplified Chinese, and another question, how can I test my translation results on the desktop terminal?

**Reply by ra3orblade** [reply]
@Xonline-Tech Currently in main localisation is switched off, in the branch I've mentioned https://github.com/anyproto/anytype-ts/pull/109/files you can either switch it in the interface or in console with `Lib.Storage.set('interfaceLang', '<lang code>')`

To update localisation files from the `l10n-anytype-ts` repository use `npm run update:locale`.

**Comment by Shampra** [comment]
Same here :).

I looked at @ra3orblade 's pulls 👍
Few questions : 
- Is it planned to have all translations (of all languages) in one file? Isn't this more cumbersome to manage than having files by language (allowing additions and updates without touching the main files)?
- Is it planned or envisaged to use a particular flow to manage translation work by the community? (My limited experience has led me to participate in other projects via Transifex, for example).

Depending on the answers: can we get started right away? After all, if it's "just" a matter of completing the text.json... I'm motivated ^^.

**Reply by fuksman** [reply]
Hey @Shampra, I think we can come to a particular decision and will "officially" launch the l18n project in the coming days. 
For now, we are considering:
-  Using multiple language files instead of a single one
-  Using https://crowdin.com/, but we will also look at Transifex, thanks
-  Streamlining the process for all platforms, not only for desktop

**Reply by ra3orblade** [reply]
There will be 1 file with string codes and english string to serve as a reference, other language files would be generated from somewhere. And there are more parts to this like errors and default objects which will be somehow localised separately.
